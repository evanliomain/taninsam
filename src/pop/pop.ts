/**
 * @module array=>array
 */
/**
 * Removes the last element from an array.
 * @return the function to apply on the array
 * @example
 * ```
 * pop<number>()([1, 2, 3, 4, 5]) // [1, 2, 3, 4]
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3, 4, 5])
 *   .chain(pop<number>())
 *   .value() // [1, 2, 3, 4]
 * ```
 */
export function pop<T>(): (array: ReadonlyArray<T>) => ReadonlyArray<T> {
  return (array: ReadonlyArray<T>) => array.slice(0, array.length - 1);
}
