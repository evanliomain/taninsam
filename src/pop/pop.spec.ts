import { describe, expect, test } from 'vitest';
import { pop } from './pop';

describe('pop function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    pop()(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3, 4, 5] |> pop === [1, 2, 3, 4]', () => {
    expect(pop()(input)).toMatchSnapshot();
  });

  test('[1] |> pop === []', () => {
    expect(pop()([1])).toMatchSnapshot();
  });

  test('[] |> pop === []', () => {
    expect(pop()([])).toMatchSnapshot();
  });
});
