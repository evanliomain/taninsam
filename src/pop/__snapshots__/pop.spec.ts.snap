// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pop function > [] |> pop === [] 1`] = `[]`;

exports[`pop function > [1, 2, 3, 4, 5] |> pop === [1, 2, 3, 4] 1`] = `
[
  1,
  2,
  3,
  4,
]
`;

exports[`pop function > [1] |> pop === [] 1`] = `[]`;

exports[`pop function > is a pure function 1`] = `
[
  1,
  2,
  3,
  4,
  5,
]
`;
